### PR TITLE
FEATURE: Save searchbar state in persitedState

### PR DESCRIPTION
**Pull request by markusguenther**
**What I did**
Replaced the local state of the component with a redux persited state.
So that the state of the searchbar is stored in the local storage like the left sidebar or right sidebar.
So the users don`t need to open it over and over again.

**How to verify it**
Open or close the search bar in the left sidebar and reload. The state should be the same.

Bildschirmfoto 2019-12-02 um 18 25 01

Resolves: #2606

### CODE REVIEW

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/index.js [hunk 1]
```diff
@@ -72,19 +79,18 @@ class NodeTreeSearchBar extends PureComponent {
...
-        const {i18nRegistry} = this.props;
-        const {searchToggled, searchValue, searchFocused, filterNodeType} = this.state;
+        const {i18nRegistry, isHiddenSearchBar} = this.props;
+        const {searchValue, searchFocused, filterNodeType} = this.state;
         const searchLabel = i18nRegistry.translate('search', 'Search', {}, 'Neos.Neos', 'Main');
 
         const searchToggleClassName = mergeClassNames({
             [style.searchToggleButton]: true,
-            [style['searchToggleButton--active']]: searchToggled
+            [style['searchToggleButton--active']]: !isHiddenSearchBar
```
Comment (dimaip) [inline packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/index.js]: Why did you reverse it? I like the old name better :)

Comment (markusguenther) [reply packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/index.js]: Can also name it like this.

File: Configuration/Settings.yaml [hunk 2]
```diff
@@ -146,6 +146,8 @@ Neos:
             isHidden: false
             contentTree:
               isHidden: true
+            searchBar:
+              searchToggled: true
```
Comment (dimaip) [inline Configuration/Settings.yaml]: Wait, but shouldn't it be hidden by default?

File: Tests/IntegrationTests/Fixtures/1Dimension/treeSearch.e2e.js [hunk 3]
```diff
@@ -16,12 +16,10 @@ test('PageTree search and filter', async t => {
     const searchmePage = 'Searchme page';
     const searchmeShortcut = 'Searchme shortcut';
 
-    const nodeTreeSearchToggler = ReactSelector('NodeTreeSearchBar IconButton');
```
Comment (dimaip) [inline Tests/IntegrationTests/Fixtures/1Dimension/treeSearch.e2e.js]: The old behaviour was actually correct. But to test that the toggles state sticks in the local storage we can actually reload amid the test run and assert that the collapsed state stayed

Comment (markusguenther) [reply Tests/IntegrationTests/Fixtures/1Dimension/treeSearch.e2e.js]: OK changed that :)

**Comment by markusguenther** [comment]
Thanks @dimaip for the feedback.

File: packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts [hunk 4]
```diff
@@ -8,13 +8,19 @@ export interface State extends Readonly<{
...
     };
 }> {}
 
 export const defaultState: State = {
     isHidden: false,
     contentTree: {
         isHidden: false
+    },
+    searchBar: {
+        searchToggled: false
```
Comment (Sebobo) [inline packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts]: Wouldn't `isHidden` fit better with the other state variables and be more explicit.
From `searchToggled` I wouldn't easily know which state means what.

**Comment by Sebobo** [comment]
It's cool we have this feature soon. I wished for it some days ago ❤️

File: packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts (hunk 4 above)
Comment (markusguenther) [reply packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts]: Had this before ;)

Comment (dimaip) [reply packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts]: The thing is that it's visible by default. Maybe 'searchShown' or 'searchVisible'?

Comment (dimaip) [reply packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts]: I dunno, maybe 'isHidden' would also work

Comment (Sebobo) [reply packages/neos-ui-redux-store/src/UI/LeftSideBar/index.ts]: `isVisible` is also fine.

**Comment by markusguenther** [comment]
And again renamed the property 🎉

**Pull request merged**